### PR TITLE
Add support for mixed and ReflectionUnionType

### PR DESCRIPTION
**Pull request by JasperS2307**

### CODE REVIEW

File: src/ReflectionParameter.php [hunk 1]
```diff
@@ -93,7 +93,7 @@ public function isPassedByReference(): bool
      *
      * @phpcsSuppress SlevomatCodingStandard.TypeHints.ReturnTypeHint
```
Comment (svandervlugt) [inline src/ReflectionParameter.php:94]: Is this phpcsSuppress still needed?

File: test/Fixtures/ReflectionReturnSelf.php [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<?php
+namespace Hostnet\Component\EntityPlugin\Fixtures;
+
+class ReflectionReturnSelf
+{
+    public function docBlock(): self
```
Comment (svandervlugt) [inline test/Fixtures/ReflectionReturnSelf.php:6]: IF this is valid now, then what is meant in src/ReflectionMethod.php  with "// Self is not valid when used in different places."

File: test/ReflectionMethodTest.php [hunk 3]
```diff
@@ -67,4 +68,14 @@ public function testGetReturnType(): void
             $this->assertEquals('array', $php7_method->getReturnType()->getName());
         }
     }
+
+    public function testGetReturnTypeSelf(): void
+    {
+        $this->assertEquals(null, $this->method->getReturnType());
+
+        if (PHP_MAJOR_VERSION >= 7) {
```
Comment (svandervlugt) [inline test/ReflectionMethodTest.php:76]: This shouldn't be needed anymore with dropped support for php 7.

File: test/Fixtures/ReflectionReturnSelf.php (hunk 2 above)
Comment (JasperS2307) [reply test/Fixtures/ReflectionReturnSelf.php:6]: I test that the function still returns null, something that was untested before

**Review by mvannes — APPROVED** [review]

File: test/ReflectionMethodTest.php [hunk 4]
```diff
@@ -67,4 +68,14 @@ public function testGetReturnType(): void
             $this->assertEquals('array', $php7_method->getReturnType()->getName());
         }
     }
+
+    public function testGetReturnTypeSelf(): void
+    {
+        $this->assertEquals(null, $this->method->getReturnType());
```
[the remark this answers is not in the document]
Comment (JasperS2307) [reply test/ReflectionMethodTest.php:72]: Will do so in a separate PR :+1:

**Review by svandervlugt — APPROVED** [review]

**Pull request merged**